### PR TITLE
update to keys metadata caching and key validation

### PR DESCRIPTION
**Pull request by henrik-me**
Update to keys metadata and key validation. Includes pseudo code to illustrate best practices.

### CODE REVIEW

File: docs/identity-platform/signing-key-rollover.md [hunk 1]
```diff
@@ -19,12 +19,69 @@ This article discusses what you need to know about the public keys that are used
...
+Our [standard libraries](reference-v2-libraries.md) implement resilient and secure caching of keys. It’s recommended to use them to avoid subtle defects in the implementation. For custom implementations, this is the rough algorithm:
+
+#### General considerations:
+* The service validating tokens should have a cache capable of storing many distinct keys (10-1000). 
+* The keys should be cached individually, using the key id (“kid” in the OIDC keys metadata specification) as a cache key.
+* The time-to-live of keys in the cache should be configured to 24 hours, with refreshes happening every hour. This makes sure the system can respond quickly to keys being removed, but has enough cache duration to not be affected by problems in fetching keys. 
+* The keys should be refreshed:
+  * Once on process startup or when cache is empty
+  * Periodically (recommended every 1 hour) as a background job 
+  * Dynamically if a token is received that has an unknown key
```
Comment (jmprieur) [inline docs/identity-platform/signing-key-rollover.md:43]:
```suggestion
  * Dynamically if a received token was signed with an unknown key (unknown **kid** or **tid** in the header)
```

File: docs/identity-platform/signing-key-rollover.md [hunk 2]
```diff
@@ -19,12 +19,69 @@ This article discusses what you need to know about the public keys that are used
...
+```
+
+#### Service Startup procedure:
+* KeyRefresh to update the keys
+* Launch a background job which will call KeyRefresh every 1h
+
+### TokenValidation procedure for validating the key (pseudo code):
+```csharp
+  Get token from input request (input token)
+  Get key id from input token
```
Comment (jmprieur) [inline docs/identity-platform/signing-key-rollover.md:71]:
```suggestion
  Get key id from input token (**kid** / **tid** header claim for JWT)
```

**Review by jmprieur — APPROVED** [review]
LGTM
I left a couple of suggestions

**Comment by henrik-me** [comment]
#sign-off

**Comment by jmprieur** [comment]
Closing. the PR was transfered to entra-docs-pr: https://github.com/MicrosoftDocs/entra-docs-pr/pull/5700

**Comment by jmprieur** [comment]
#please-close

**Pull request closed without merging**